### PR TITLE
Remote Docker Execution

### LINKED ISSUE #163: Add Docker Remote Execution
### Context
In order to facilitate development on cloud infrastructure it would be nice to provide the ability to execute all Docker commands on a remote host.

### Feature
Add a field to the root of the skelebot.yaml config called `host` to allow the user to set the host server for docker execution.
Add a field to the root of the job object in the skelebot.yaml config called `host` to allow the user to set the host server for a specific job.
Add an optional global parameter called `host` to allow the user to set the host server for a specific command.
The job `host` value would override the global `host` value and the global parameter would override any job or global `host` value that is set.

The Docker Host can be set with: `export DOCKER_HOST=ssh://user@your_server_ip`
or `docker --host ssh://user@your_server_ip ...`

---

### Acceptance Criteria
- [x] Add Global Host Config Field
- [x] Add Job Host Config Field
- [x] Add Global Host Parameter
- [x] Bump Minor Version
- [x] Update Changelog
- [x] Add/Update Documentation

### PR DESCRIPTION
**Pull request by sshookman**
Resolves #163

### Note
This was way too complicated for something seemingly so simple. We need to refactor the code for the Docker System. It should probably be a class with a string builder, that way we can just change certain properties (like host) in one place and not have to worry about it.

### Solution
I refactor the way we build the docker commands to make use of the builder pattern. I ultimately left the host as an optional parameter to the initial build instead of a separate function in order to prevent a lot of complicated if statements and convoluted code.

This introduces three ways to set the Docker Host:
1. Global "host" field in the root of the skelebot.yaml
2. Job level "host" field in the skelebot.yaml
3. Command line "host" parameter

The job level field takes precedence over the global field, and the command line parameter takes precedence over the job-level field.

This allows users to specify a Docker Host on another machine for running their jobs and various other tasks.

Bonus: this may be of great help to Windows users who can now send their code to another machine to run it.

### CODE REVIEW

**Comment by sshookman** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- skelebot/systems/execution/dockerCommand.py  2
         

Complexity decreasing per file
==============================
+ skelebot/systems/execution/docker.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/skelebot/skelebot/pullRequest?prid=5938912&bid=19467938)

**Comment by addhyan** [comment]
We should probably add Tuo in here as well.

**Comment by sshookman** [comment]
I couldn't find him on the list when I typed his name. What is his username?

File: docs/docker-host.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Skelebot offers the ability to run your docker commands on another host aside from your local
+machine. To specify a different host for the entire project, simply add the `host` field to the
+root of your skelebot.yaml config file.
+
+```
+host: ssh://root@remote.host
+```
+
+The same parmeter is available at the job-level, and will override the global field if specified.
```
Comment (jagmoreira) [inline docs/docker-host.md:15]:
```suggestion
The same parameter is available at the job-level, and will override the global field if specified.
```

File: test/test_systems_execution_docker.py [hunk 2]
```diff
@@ -164,9 +164,24 @@ def test_build_non_ephemeral(self, mock_getcwd, mock_call, mock_expanduser):
...
+    def test_build_host(self, mock_getcwd, mock_call, mock_expanduser):
+        folderPath = "{path}/test/files".format(path=self.path)
+
+        mock_expanduser.return_value = "{path}/test/plugins".format(path=self.path)
+        mock_getcwd.return_value = folderPath
+        mock_call.return_value = 0
+
+        config = sb.systems.generators.yaml.loadConfig()
+
+        sb.systems.execution.docker.build(config, "root@hosty.McHostface")
```
Comment (jagmoreira) [inline test/test_systems_execution_docker.py:182]: best hostname ever?

Comment (sshookman) [reply test/test_systems_execution_docker.py:182]: I thought so

**Comment by sshookman** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- skelebot/systems/execution/dockerCommand.py  2
         

Complexity decreasing per file
==============================
+ skelebot/systems/execution/docker.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/skelebot/skelebot/pullRequest?prid=5938912&bid=19467938)

**Pull request merged**